csvapi/parser.py

```diff
@@ -39,9 +51,16 @@ def to_sql(table, urlhash, storage):
 
 
 def parse(filepath, urlhash, storage, encoding=None, sniff_limit=SNIFF_LIMIT):
-    if is_binary(filepath):
+    if is_xls(filepath):
```

Comment:
Maybe it's time to do something smarter: instead of executing `file` three times for a csv file, maybe make a `detect_type` function that returns the detected type based on one run of `file`

Reply:
And the `from_excel`func can go away, not very useful.